tests/test_install.py

```diff
@@ -264,6 +264,50 @@ def test_pip_args_with_windows_path(pipx_temp_env, capsys):
     assert r"D:\\TEST\\DIR" in captured.err
 
 
+@pytest.mark.parametrize("constraint_flag", ["-c ", "--constraint ", "--constraint="])
+def test_pip_args_with_constraint_relative_path(constraint_flag, pipx_temp_env, tmp_path, caplog):
+    constraint_file_name = "constraint.txt"
+    pycowsay_version = "0.0.0.1"
+
+    subprocess_pycowsay_available_versions = subprocess.run(
+        [sys.executable, "-m", "pip", "index", "versions", "pycowsay"], capture_output=True, text=True, check=False
+    )
+    subprocess_pycowsay_available_versions_output = subprocess_pycowsay_available_versions.stdout
+    output_lines = subprocess_pycowsay_available_versions_output.split("\n")
+    available_versions_line = next(line for line in output_lines if "Available versions:" in line)
+    versions_string = available_versions_line.split(":")[1].strip()
+    available_versions = versions_string.split(", ")
+    assert pycowsay_version in available_versions
+
+    os.chdir(tmp_path)
+    current_directory = Path(".").expanduser().resolve()
+
+    temp_file = current_directory / constraint_file_name
+    temp_file.touch()
+    temp_file.write_text(f"pycowsay=={pycowsay_version}")
```

Comment:
We have a small database for our test packages, which we use in tests.

```suggestion
    temp_file.write_text(PKG["pycowsay"]["spec"])
```

Reply:
@guysalt I think you overlooked this comment?

Reply:
Also here, if someone change to the newer version the test will not check if the older versions indeed installed.

**Changed**